### PR TITLE
MGMT-19012: The controller shouldn't remove the finalizer as long the dataImage exists

### PR DESCRIPTION
**Pull request by eranco74**
The finalizer will be removed by the controller if the referenced BareMetalHost or the DataImage no longer exists.

### CODE REVIEW

File: controllers/imageclusterinstall_controller.go [hunk 1]
```diff
@@ -697,33 +697,27 @@ func (r *ImageClusterInstallReconciler) getBMH(ctx context.Context, bmhRef *v1al
...
 
-func (r *ImageClusterInstallReconciler) removeBMHDataImage(ctx context.Context, log logrus.FieldLogger, bmh *bmh_v1alpha1.BareMetalHost, bmhRef *v1alpha1.BareMetalHostReference) error {
-	dataImage := &bmh_v1alpha1.DataImage{}
-
-	key := types.NamespacedName{
-		Name:      bmhRef.Name,
-		Namespace: bmhRef.Namespace,
+func (r *ImageClusterInstallReconciler) removeBMHDataImage(ctx context.Context, log logrus.FieldLogger, bmhRef types.NamespacedName) (*bmh_v1alpha1.DataImage, error) {
+	dataImage, err := r.deleteDataImage(ctx, log, bmhRef)
+	if err != nil || dataImage == nil {
```
Comment (carbonin) [inline controllers/imageclusterinstall_controller.go:702]: If the data image was already removed (`err == nil && dataImage == nil`) should we still attach and reboot the BMH? Looks like we're exiting early in that case here.

**Comment by eranco74** [comment]
/hold
I'd like to run the e2e with [this](https://github.com/openshift/release/pull/57011) change

File: controllers/imageclusterinstall_controller.go (hunk 1 above)
Comment (eranco74) [reply controllers/imageclusterinstall_controller.go:702]: The only reason we attach and reboot is to allow the dataImage to be deleted.
If the dataImage doesn't exist we don't want to touch the BMH.

Comment (carbonin) [reply controllers/imageclusterinstall_controller.go:702]: Ah okay

**Review by carbonin — APPROVED** [review]
Looks good to me, unhold when you're ready @eranco74

**Pull request merged**